pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/MaxAggregationFunction.java

```diff
@@ -92,6 +93,15 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
         aggregationResultHolder.setValue(Math.max(max, aggregationResultHolder.getDoubleResult()));
         break;
       }
+      case BIG_DECIMAL: {
+        BigDecimal[] values = blockValSet.getBigDecimalValuesSV();
+        BigDecimal max = values[0];
+        for (int i = 0; i < length & i < values.length; i++) {
+          max = values[i].max(max);
+        }
+        aggregationResultHolder.setValue(Math.max(max.doubleValue(), aggregationResultHolder.getDoubleResult()));
```

Comment:
Suggest adding a comment (or TODO) stating even though the source data has BIG_DECIMAL type, we still only support double precision. Same for other functions

Reply:
will do